vtex-sites/base.store
# feat: Integrate Out of Stock with regionalization

## What's the purpose of this pull request?

Whenever the user provides his/her location/postal code, the availability for the product will be calculated based on the informed location. This PR just check if the `postalCode` is set in our `session`, if it is the case, the _Out of Stock_ title will be changed to _Unavailable in Your Location_.  

## How does it work?

|Default title|Title when postalCode is informed|
|-|-|
|<img width="404" alt="image" src="https://user-images.githubusercontent.com/10616243/161318211-05c856bb-718d-478a-b741-c82b117a1159.png">|<img width="407" alt="image" src="https://user-images.githubusercontent.com/10616243/161320837-6f9c39a0-80cb-4dda-add6-3b1f75deddaa.png">|

## How to test it?

One way you can see the component in the page is opening the PDP of a product that is unavailable (which you probably won't find in Base.Store) or you can remove the `!` from the `!availability &&` in line 179 of `ProductDetails.tsx` file (just for testing purposes). 

Also remember to set the postalCode:
<img width="250" alt="image" src="https://user-images.githubusercontent.com/10616243/161321174-9926f1f9-2914-48ba-a0bb-024ee93c05be.png">


## References

Related [Jira task](https://vtex-dev.atlassian.net/browse/FSSS-235?atlOrigin=eyJpIjoiOGUyMDNkMWI2ZThjNDgzMGFhMGZkMDExY2FhOGVmOTgiLCJwIjoiaiJ9)
Related [Figma file](https://www.figma.com/file/u9z8sTWqBm8eVGsazPLlnX/Handoff-v4-(Regionalization)?node-id=604%3A159449)
